src/Workspaces/Core/Portable/Storage/AbstractPersistentStorageService.cs

```diff
@@ -35,44 +35,43 @@ protected AbstractPersistentStorageService(IPersistentStorageLocationService loc
         protected abstract string GetDatabaseFilePath(string workingFolderPath);
 
         /// <summary>
-        /// Can throw.  If it does, the caller (<see cref="CreatePersistentStorage"/>) will attempt
+        /// Can throw.  If it does, the caller (<see cref="CreatePersistentStorageAsync"/>) will attempt
         /// to delete the database and retry opening one more time.  If that fails again, the <see
         /// cref="NoOpPersistentStorage"/> instance will be used.
         /// </summary>
-        protected abstract IChecksummedPersistentStorage? TryOpenDatabase(SolutionKey solutionKey, Solution? bulkLoadSnapshot, string workingFolderPath, string databaseFilePath);
+        protected abstract ValueTask<IChecksummedPersistentStorage?> TryOpenDatabaseAsync(SolutionKey solutionKey, Solution? bulkLoadSnapshot, string workingFolderPath, string databaseFilePath);
         protected abstract bool ShouldDeleteDatabase(Exception exception);
 
         IPersistentStorage IPersistentStorageService.GetStorage(Solution solution)
```

Comment:
💡 If we are anticipating `GetStorageAsync` going to the platform (on any path), this method need to use JTF for the sync-to-async boundary or it will deadlock.

Reply:
ah.  will do that.